Source/OCMock/OCMMacroState.m

```diff
@@ -45,7 +45,10 @@ + (OCMStubRecorder *)endStubMacro
 	if([recorder wasUsed] == NO)
 	{
 		[NSException raise:NSInternalInconsistencyException
-					format:@"Mock object was not used in OCMStub/OCMExpect/OCMReject. Did you accidentally use a real object?"];
+					format:@"Did not record an invocation in OCMStub/OCMExpect/OCMReject.\n"
+						   @"Possible causes are:\n"
+						   @"- The target is not a mock object.\n"
+						   @"- The method is directly implemented by NSObject. Use `[[[foo stub] andReturn:] METHOD]` instead of macro."];
```

Comment:
This is incorrect. It is possible to stub methods on `NSObject` with a macro. The following test passes:

```
- (void)testCanStubAndThenVerifySomeInstanceMethodsOnNSObject
{
    id mock = OCMClassMock([NSObject class]);

    OCMStub([mock classForArchiver]).andReturn([NSString class]);

    XCTAssertEqualObjects([NSString class], [mock classForArchiver], @"Should have stubbed method");
}
```

Instance methods on `NSObject` are not automatically replaced, which means you can't just verify them. However, it is possible to stub some instance methods. And after that you can even verify that they were invoked.

Reply:
Sorry about that. I jumped to conclusions. I think the updated text is more accurate as to exactly what is happening.